### DISCUSSION TITLE
Change the order on the /documents route output

### DISCUSSION THREAD

**Discussion by gorkagv** [meilisearch/product#103] [Feedback & Feature Proposal]
Hi.

Currently when doing an index search (/ indexes /: index_uid / documents), the documents are sorted by primaryKey by default, in my index the primaryKey is "eId".

{
    "eId": 31,
    "name": "Charles”,
..
},
{
    "eId": 32,
    "name": "Ingrid”,
..
},
{
    "eId": 33,
    “name”: “Robert”,
..
},
{
    "eId": 34,
    “name”: "Garcia”,
..
},

Is there a way to sort the index by multiple attributes (eg name), instead of the default sort?

I have added a ranking rule using “/ indexes /: index_uid / settings / ranking-rules”:
"typo", "words", "proximity", "attribute", "wordsPosition", "exactness", "desc(name)"
but the results are still sorted by the primaryKey instead of the “name” attribute.

I have rebuilt the index using the commands “php artisan scout:flush “Model”” and “php artisan scout:import “Model””

I have also tried this: "typo", "desc(name)”, “words", "proximity", "attribute", "wordsPosition", "exactness" using “/ indexes /: index_uid / settings / ranking-rules” but the results are still sorted by the primary key.

**Comment by bidoubiwa** [comment]
Hello @gorkagv! Ranking rules are not applied on `/indexes/:uid/documents`. But they are on the placeholdersearch that may be a solution if you are trying to provide default documents to your user. 

Here is the related documentation: https://docs.meilisearch.com/reference/features/search_parameters.html#query-q
I hope this is what you ask!

**Comment by gorkagv** [comment]
I am using “meilisearch / meilisearch-vue” like in this example -> https://codesandbox.io/s/ms-vue-is-1d6bi?fontsize=14&hidenavigation=1&theme=dark
I need to sort the results by an attribute, for example by date, newests result first, or by name, results sorted alphabetically.

In the documentation I have found this in the ranking rules section (https://docs.meilisearch.com/reference/api/ranking_rules.html#update-ranking-rules):

```
_An array that contain ranking rules sorted by order of importance.
To add your own ranking rule, you have to communicate either asc for ascending order or desc for descending order followed by the field name in brackets.
	•	To apply an ascending sorting (results sorted by increasing value of the attribute): asc(attribute_name) 
	•	To apply a descending sorting (results sorted by decreasing value of the attribute): desc(attribute_name) 
curl \
  -X POST 'http://localhost:7700/indexes/movies/settings/ranking-rules' \
  --data '[
      "typo",
      "words",
      "proximity",
      "attribute",
      "wordsPosition",
      "exactness",
      “desc(date)",
      “asc(name)”,
  ]'_
```

but when I do some search or get all results (without doing any search) the results are not sorted by the attributes “date” or “name”, the results are sorted by PrimaryKey.

How could I sort the results by those attributes?.

Thanks in advance.

**Comment by bidoubiwa** [comment]
Hey @gorkagv as this is a issue linked to meilisearch/meilisearch I will move your issue.

**Comment by curquiza** [comment]
Hello @gorkagv can you send us your dataset and your detailed steps so that we can reproduce your search?

**Comment by gorkagv** [comment]
For privacy reasons I cannot send the dataset.

But in this example dataset -> https://codesandbox.io/s/ms-vue-is-1d6bi?fontsize=14&hidenavigation=1&theme=dark

How could I sort the results list by date or alphabetically?

I have tried adding two parameters (“desc(date)",“asc(name)”) to the ranking rules in my dataset, but the results are not ordered by those parameters.

Thanks in advance.

**Comment by curquiza** [comment]
Hello @gorkagv, I've seen you linked the dataset we are using for our demo: all the fields have string values and the custom ranking rules (`asc`/`desc`) does not work with strings

<img width="886" alt="Capture d’écran 2021-03-24 à 19 31 28" src="https://user-images.githubusercontent.com/20380692/112365775-53b30d80-8cd8-11eb-8c7a-179a4e13117b.png">

The docs link: https://docs.meilisearch.com/reference/features/settings.html#custom-ranking-rule

If it's the case for your dataset, it's why it does not work.
For the `date` field, you can transform your date into timestamp (these would be integers) and then do your ranking rule with this timestamp field.
For the `name` field, I don't have any workarounds to suggest for the moment. Except maybe add a field that would represent the position of your document ranked by alphabetical number and then, create a ranking rule on this field. But this solution does not work if you add documents on MeiliSearch on regular basis.

FYI, we are considering allowing the strings in the custom ranking, but you have to be aware this would impact the performance of the search.
